packages/@icon-magic/generate/src/plugins/svg-generate.ts

```diff
@@ -24,38 +24,103 @@ import * as fs from 'fs-extra';
 import * as path from 'path';
 import Svgo from 'svgo';
 
-// TODO: check if we should add more properties here
 export interface SvgGenerateOptions {
-  propCombo: object;
-  addSupportedDps?: boolean;
-  isColoredIcon?: boolean;
-  setSize?: boolean;
-  setViewbox?: boolean;
+  propCombo?: object;
+  addSupportedDps?: 'all' | 'current' | 'none'; // when set to current, only adds the current size. If not, defaults to adding all allowed sizes
+  isColored?: boolean; // if this is set, the fill of the icon isn't updated but respected
+  isFixedDimensions?: boolean; // if this is true, then adds a width and height attribute to the svg (defaults to true) and no viewBox
+  colorByNameMatching?: string[]; // if this is set to true, then set isColored only if the name of the flavor contains color in it
 }
 
 export const svgGenerate: GeneratePlugin = {
   name: 'svg-generate',
-  fn: async (flavor: Flavor, icon: Icon, _params?: object): Promise<Flavor> => {
+  fn: async (
+    flavor: Flavor,
+    icon: Icon,
+    params?: SvgGenerateOptions
+  ): Promise<Flavor> => {
+    // build the attributes object that contains attributes to be added to the svg
+    const attributes = { id: `${icon.iconName}-${flavor.name}` };
+    let setCurrentColor = true; // by default, sets the colour of the icon to take the currentColor
+    const flavorName: string = path.basename(flavor.name);
+
+    if (params) {
+      let dataSupportedDps;
+      switch (params.addSupportedDps) {
+        case 'current':
+          // get the mapping object from the metadata
+          const nameSizeMapping =
+            icon.metadata && icon.metadata.nameSizeMapping;
+
+          // if the metadata doesn't contain nameSizeMapping, throw an error
+          if (!nameSizeMapping) {
+            throw new Error(
```

Comment:
add try, catch.

Reply:
Adding the try catch within the plugin manager so that execution on the rest of the flavors can continue.